cli/command/registry/login.go

```diff
@@ -39,14 +41,62 @@ func NewLoginCommand(dockerCli command.Cli) *cobra.Command {
 
 	flags.StringVarP(&opts.user, "username", "u", "", "Username")
 	flags.StringVarP(&opts.password, "password", "p", "", "Password")
+	flags.StringVarP(&opts.passwordFile, "password-file", "f", "", "Password file whose contents are the password itself")
 
 	return cmd
 }
 
+/* getPasswordFromFile reads the contents of the file, or stdin if the file is
+ * == "-".
+ *
+ * It also trims off the last \n in the file, if it exists. Most people don't
+ * have \ns in their password, and this allows stuff like echo "password" >
+ * foo, without having to remember to pass -n, or vi, which can be configured
+ * to automatically append newlines, etc.
+ *
+ * For users that do have a \n as the last character of their password, they
+ * need to store it as \n\n. I think this conforms to the principle of least
+ * surprise, but I could be wrong :)
+ */
```

Comment:
I believe the go convention is to use `//` single line comments for everything instead of `/* ... */`